crates/node-core/src/node_config.rs

```diff
@@ -1452,6 +1488,36 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_node_exit_future_terminate_true() {
+        let (tx, rx) = oneshot::channel::<Result<(), BeaconConsensusEngineError>>();
+
+        let _ = tx.send(Ok(()));
+
+        let node_exit_future = NodeExitFuture::new(rx, true);
+
+        let res = node_exit_future.await;
+
+        assert!(res.is_ok());
+    }
+
+    use std::time::Duration;
+    use tokio::time::timeout;
+    #[tokio::test]
+    async fn test_node_exit_future_terminate_false() {
+        let (tx, rx) = oneshot::channel::<Result<(), BeaconConsensusEngineError>>();
+
+        let _ = tx.send(Ok(()));
+
+        let node_exit_future = NodeExitFuture::new(rx, false);
+        match timeout(Duration::from_millis(100), node_exit_future).await {
+            Ok(_) => panic!("terminations flag was set to FALSE - future shouldn't be resolved"),
+            Err(_) => {
+                println!("termination flag was set to FALSE - future expectadly didn't resolve")
+            }
+        }
```

Comment:
pedantic nit:

this check can be done via `poll_fn(|cx| node_exit_future.poll(x))` so we don't need the timeout here